src/js/select.js

```diff
@@ -227,22 +228,29 @@ Menu.prototype._createMenuEl = function(wrapperEl, selectEl) {
       // add row item to menu
       rowEl = doc.createElement('div');
       rowEl.textContent = loopEl.textContent;
-      rowEl._muiIndex = indexNum;
-
-      // handle selected options
-      if (loopEl.selected) {
-        rowEl.className = selectedClass;
-        selectedRow = menuEl.children.length;
-      }
 
       // handle optgroup options
       if (inGroup) jqLite.addClass(rowEl, 'mui-optgroup__option');
 
-      menuEl.appendChild(rowEl);
+      if (loopEl.disabled) {
+        // do not attach muiIndex to disable <option> elements to make them
+        // unselectable.
+        rowEl.className = disabledClass;
+      } else {
+        rowEl._muiIndex = indexNum;
+
+        // handle selected options
+        if (loopEl.selected) {
+          rowEl.className = selectedClass;
+          selectedRow = menuEl.children.length;
+        }
+      }
 
       // add to index map
       indexMap[indexNum] = rowEl;
```

Comment:
Thanks for pointing out the typo, `index` should be `_muiIndex` in `selectCurrent()`.

If `itemArray` only consists of menu rows corresponding to enabled options then I think the above code should work. Why would you need to loop through the disabled options in the inc/dec methods?